clients/src/main/java/org/apache/kafka/common/record/MemoryRecords.java

```diff
@@ -160,20 +160,16 @@ public FilterResult filterTo(TopicPartition partition, RecordFilter filter, Byte
     private static FilterResult filterTo(TopicPartition partition, Iterable<MutableRecordBatch> batches,
                                          RecordFilter filter, ByteBuffer destinationBuffer, int maxRecordBatchSize,
                                          BufferSupplier decompressionBufferSupplier) {
-        long maxTimestamp = RecordBatch.NO_TIMESTAMP;
-        long maxOffset = -1L;
-        long shallowOffsetOfMaxTimestamp = -1L;
-        int messagesRead = 0;
-        int bytesRead = 0; // bytes processed from `batches`
-        int messagesRetained = 0;
-        int bytesRetained = 0;
-
+        FilterResult filterResult = new FilterResult();
         ByteBufferOutputStream bufferOutputStream = new ByteBufferOutputStream(destinationBuffer);
+        int totalBytesRead = 0;
```

Comment:
Since `FilterResult` is mutable, could we just increment the `bytesRead` and `messagesRead` fields directly?